datadog_checks_base/datadog_checks/base/checks/win/winpdh_base.py

```diff
@@ -105,6 +105,41 @@ def __init__(self, name, init_config, agentConfig, instances, counter_list):
         if key is None or not self._metrics.get(key):
             raise AttributeError('No valid counters to collect')
 
+    def _get_netresource(self, remote_machine):
+        # To connect you have to use the name of the server followed by an administrative share.
+        # Administrative shares are hidden network shares created that allow system administrators to have remote access
+        # to every disk volume on a network-connected system.
+        # These shares may not be permanently deleted but may be disabled.
+        # Administrative shares cannot be accessed by users without administrative privileges.
+        #
+        # This page explains how to enable them: https://www.wintips.org/how-to-enable-admin-shares-windows-7/
+        #
+        # The administrative share can be:
+        # * A disk volume like c$
+        # * admin$: The folder in which Windows is installed
+        # * fax$: The folder in which faxed pages and cover pages are cached
+        # * ipc$: Area used for interprocess communication and is not part of the file system.
+        # * print$: Virtual folder that contains a representation of the installed printers
+        # * Domain controller shares: Windows creates two domain controller specific shares called sysvol and netlogon
+        # which do not have $ appended to their names.
+        administrative_share = self.instance.get('admin_share', DEFAULT_SHARE)
+
+        nr = win32wnet.NETRESOURCE()
+
+        # Specifies the network resource to connect to.
+        nr.lpRemoteName = r"\\%s\%s" % (remote_machine, administrative_share)
+
```

Comment:
Let's add `nr.rstrip('\\')` to remove any trailing backslash. Especially useful with `admin_share: ""`

Reply:
Connection to `\\<HOSTNAME>` can also work, but `\\<HOSTNAME>\` will fail

Reply:
`nr.rstrip('\\')`or `nr.lpRemoteName.rstrip('\\')`?

Reply:
Yeah I meant `nr.lpRemoteName.rstrip('\\')`